src/sth_server.js

```diff
@@ -77,95 +77,194 @@
           var databaseName = sthDatabase.getDatabase(
             request.headers['fiware-service']);
 
-          // Compose the collection name for the required data
-          var collectionName = sthDatabase.getCollectionName4Aggregated(
-            request.headers['fiware-servicepath'],
-            request.params.entityId,
-            request.params.entityType,
-            request.params.attrName
-          );
+          if ((request.query.lastN || request.query.lastN === 0)  ||
+            ((request.query.hLimit || request.query.hLimit === 0) &&
+            (request.query.hOffset || request.query.hOffset === 0))) {
+            // Raw data is requested
+
+            // Compose the collection name for the required data
+            var collectionName = sthDatabase.getCollectionName4Events(
```

Comment:
Agreed! Really nice proposal... ;) Thanks! Fixed in https://github.com/telefonicaid/IoT-STH/commit/2300b698793e6bc3b1ed46cda6646eb6375ac971